### PR TITLE
Added pendingDot to Timeline

### LINKED ISSUE #9514: Timeline pending dot
### What problem does this feature solve?
Customize timeline pending dot when pending is not false.

### What does the proposed API look like?
``` javascript
<Timeline
  pending={boolean|string|ReactNode}
  pendingDot={string|ReactNode}
>
</Timeline>
```

### PR DESCRIPTION
**Pull request by iannbing**
Fix #9514

Added pendingDot to Timeline. One can provide a React component to replace the default pending dot.

### CODE REVIEW

File: components/timeline/demo/pending.md [hunk 1]
```diff
@@ -7,11 +7,11 @@ title:
 
 ## zh-CN
 
-当任务状态正在发生，还在记录过程中，可用幽灵节点来表示当前的时间节点。（用于时间正序排列）
+当任务状态正在发生，还在记录过程中，可用幽灵节点来表示当前的时间节点（用于时间正序排列）。当pending值为false，可用定制元件替换默认时间图点。
```
Comment (afc163) [inline components/timeline/demo/pending.md]: 中英文之间要加空格。

Comment (SimpleFrontend) [reply components/timeline/demo/pending.md]: Updated

File: components/timeline/Timeline.tsx [hunk 2]
```diff
@@ -33,7 +35,7 @@ export default class Timeline extends React.Component<TimelineProps, any> {
     const pendingItem = (!!pending) ? (
       <TimelineItem
         pending={!!pending}
-        dot={<Icon type="loading" />}
+        dot={pendingDotNode}
```
Comment (yutingzhao1991) [inline components/timeline/Timeline.tsx]: 似乎直接在这里写 `doc={pendingDot || <Icon type="loading" />}` 简洁一点，逻辑也更清楚一点。

Comment (SimpleFrontend) [reply components/timeline/Timeline.tsx]: Make sense. Updated

**Review by yutingzhao1991 — APPROVED** [review]

**Pull request merged**